internal/batches/executor/coordinator.go

```diff
@@ -221,6 +224,18 @@ func (c *Coordinator) loadCachedStepResults(ctx context.Context, task *Task, glo
 	return nil
 }
 
+func writeToCache(ctx context.Context, cache cache.Cache, stepResult execution.AfterStepResult, task *Task) error {
+	globalEnv := os.Environ()
+	cacheKey := task.cacheKey(globalEnv)
```

Comment:
I'm not particularly concerned about this, but if we wanted to avoid the `os.Environ()` on each task, we could grab it in `NewCoordinator` and calculate the cache key in the `WriteToCache` callback before invoking this function.

Not worth changing unless we're back in here — `os.Environ` isn't free, but it's probably not expensive enough (one read lock and _n_ string copies) to prematurely optimise. Just something to keep in mind for the future.

Reply:
I think that's a good call actually. Doesn't seem super hard to do, @BolajiOlajide wdyt about that?

Reply:
Hey! Just catching up with this.
Yes, it's non-trivial and a non-breaking change so I can do it. It's awesome because it also reduces repetition in all tasks.

I'll update accordingly.